packages/cli/src/deploy/ensureDeployer.ts

```diff
@@ -1,16 +1,15 @@
-import { Account, Address, Chain, Client, Transport } from "viem";
+import { Account, Address, Chain, Client, Transport, sliceHex } from "viem";
 import { getBalance, getBytecode, sendRawTransaction, sendTransaction, waitForTransactionReceipt } from "viem/actions";
 import deployment from "./create2/deployment.json";
 import { debug } from "./debug";
 
 const deployer = `0x${deployment.address}` as const;
-const deployerBytecode = `0x${deployment.bytecode}` as const;
 
 export async function ensureDeployer(client: Client<Transport, Chain | undefined, Account>): Promise<Address> {
   const bytecode = await getBytecode(client, { address: deployer });
   if (bytecode) {
     debug("found CREATE2 deployer at", deployer);
-    if (bytecode !== deployerBytecode) {
+    if (bytecode !== sliceHex(`0x${deployment.creationCode}`, 14)) {
```

Comment:
another approach here could be
```suggestion
    if (deployment.creationCode.endsWith(bytecode.slice(2))) {
```
but that feels less accurate to me